include/zephyr/net/phy.h

```diff
@@ -176,7 +182,8 @@ __subsystem struct ethphy_driver_api {
 	int (*get_link)(const struct device *dev, struct phy_link_state *state);
 
 	/** Configure link */
-	int (*cfg_link)(const struct device *dev, enum phy_link_speed adv_speeds);
+	int (*cfg_link)(const struct device *dev, enum phy_link_speed adv_speeds,
```

Comment:
`adv_speeds` now has an implicit dual meaning: either a list of advertised speeds when auto-negotiation is enabled, or a forced mode when auto-negotiation is disabled. That overload is a little awkward.

What do you think about adding a check in `phy_configure_link()` that returns `-EINVAL` if auto-negotiation is disabled and more than one speed/duplex mode is specified?

Reply:
Sure